dali/test/python/test_operator_readers_webdataset_corner.py

```diff
@@ -0,0 +1,188 @@
+
+
+def test_single_sample_and_junk():
+    test_batch_size = 1
+    num_samples = 1
+    tar_file_path = get_dali_extra_path() + "/db/webdataset/sample-tar/single_junk.tar"
+    index_file = generate_temp_index_file(tar_file_path)
+
+    extract_dir = generate_temp_extract(tar_file_path)
+    equivalent_files = list(sorted(glob(extract_dir.name + "/*")))
+
+    compare_pipelines(
+        webdataset_raw_pipeline(
+            tar_file_path,
+            index_file.name,
+            ["txt"],
+            batch_size=test_batch_size,
+            device_id=0,
+            num_threads=1,
+        ),
+        file_reader_pipeline(
+            equivalent_files, ["txt"], batch_size=test_batch_size, device_id=0, num_threads=1
+        ),
+        test_batch_size,
+        math.ceil(num_samples / test_batch_size) * 10,
+    )
+    wds_pipeline = webdataset_raw_pipeline(
+        tar_file_path,
+        index_file.name,
+        ["txt"],
+        batch_size=test_batch_size,
+        device_id=0,
+        num_threads=1,
+    )
+    wds_pipeline.build()
+    assert_equal(list(wds_pipeline.epoch_size().values())[0], num_samples)
+
+
+def test_wide_sample():
+    test_batch_size = 1
+    num_samples = 1
+    tar_file_path = get_dali_extra_path() + "/db/webdataset/sample-tar/wide.tar"
+    index_file = generate_temp_index_file(tar_file_path)
+
+    extract_dir = generate_temp_extract(tar_file_path)
+    equivalent_files = list(
+        sorted(glob(extract_dir.name + "/*"))
+    )
+
+    num_components = 1000
+    compare_pipelines(
+        webdataset_raw_pipeline(
+            tar_file_path,
+            index_file.name,
+            [str(x) for x in range(num_components)],
+            batch_size=test_batch_size,
+            device_id=0,
+            num_threads=1,
+        ),
+        file_reader_pipeline(
+            equivalent_files,
+            [str(x) for x in range(num_components)],
+            batch_size=test_batch_size,
+            device_id=0,
+            num_threads=1,
+        ),
+        test_batch_size,
+        math.ceil(num_samples / test_batch_size) * 10,
+    )
+    wds_pipeline = webdataset_raw_pipeline(
+        tar_file_path,
+        index_file.name,
+        ["txt"],
+        batch_size=test_batch_size,
+        device_id=0,
+        num_threads=1,
+    )
+    wds_pipeline.build()
+    assert_equal(list(wds_pipeline.epoch_size().values())[0], num_samples)
+
+
+def test_mmap_dtype_incompatibility():
+    assert_raises(RuntimeError, corner_case, dtypes=[dali.types.INT8, dali.types.FLOAT64])
+
+
+def test_dont_use_mmap():
+    corner_case(dont_use_mmap=True)
+
+
+def test_skip_cached_images():
+    corner_case(skip_cached_images=True)
+
+
+def test_pad_last_batch():
+    corner_case(pad_last_batch=True)
+
+
+def test_lazy_init():
+    corner_case(lazy_init=True)
+
+
+def test_read_ahead():
+    corner_case(read_ahead=True)
+
+
+def test_errors():
+    pass
```

Comment:
?

Reply:
forgot about that one, implemented it now